src/components/PlaylistPage/LessonItem.js

```diff
@@ -12,11 +12,51 @@ import LevelIcon from '../LevelIcon';
 import {getTranslator} from '../../selectors/translate';
 import {getLessonIntro} from '../../util';
 
+const percentObject = {};
+const getCheckboxProgress = (path, checkboxes) => {
```

Comment:
getCheckboxProgress er en relativt kostbar funksjon, og den blir kalt på hver render av alle LessonItems. Jeg foreslår at dette flyttes ut i en selector (eller gjerne 2 selectorer, getNumberOfCheckedCheckboxes og getTotalNumberOfCheckboxes).

Fordi du må sende med props (path) og fordi den brukes av mer enn en komponent, så må man ta høyde for dette i memoizeringen: Se https://github.com/reactjs/reselect#sharing-selectors-with-props-across-multiple-components
(evt bruk re-reselect: https://github.com/toomuchdesign/re-reselect)